api/bots/poll/poll.py

```diff
@@ -0,0 +1,266 @@
+            It currently keeps a running count of the votes.
+
+           Design:
+             - polls are visible only within streams;
+             - one poll per user (per topic, per stream);
+             - private messages (or in the stream) control the bot;
+             - the bot only responds in a stream by posting a new or updated poll
+               (and otherwise replies to the sender, eg. for confirmations/errors)
+             - commands:
+                 * new: adds/resets the poll on the stream/topic, with a given title
+                   text, followed by each poll option on a separate line, eg.:
+                     new [stream topic]
+                     How do you feel about this bot?
+                     Terrible!
+                     It's OK.
+                     Great!
+                   This generates a poll with 3 options, defaulting with the current
+                   stream/topic, or using those specified
+                 * vote: adds a vote in a given poll from yourself; you cannot
+                   vote more than once in a given poll. Example command:
+                     vote 99 3 [stream topic]
+                   If a valid index (eg. 1-3 above), the tally for that option is increased,
+                   and the original poll message updated; in this case, poll 99 in the current
+                   stream and topic (unless specified)
+                 * end: ends a poll that you own, allowing you to start a new one, eg.
+                     end [stream topic]
+           BUGS/LIMITATIONS/IDEAS:
+             - *** Currently, stream+topic are space-separated, which rules out spaces in each
+             - When do polls end? What if the poll text can't be edited (a server setting?)
+             - Polls are only in streams - sufficient?
+             - One poll per user per topic - sufficient?
+             - (Once complete, this could migrate to the command-bot API)
+             - Improve help text generally
+             - Support changing votes?
+             - Show votes only at the end?
+             - Other vote 'schemes'?
+             - External vote engines?
+             - External vote storage? (current state is 'maintained' in the message too!)
+            '''
+
+    def handle_message(self, message, bot_handler, state_handler):
+        help_content = "\nCommands: help, new, vote, end"
+
+        content = message["content"]
+        sender = message["sender_email"]
+        sender_id = message["sender_id"]
+
+        # Break down the text supplied into potential input sets
+        lines = content.split('\n')
+        main_line = lines[0].split(' ')
+        command = main_line[0]
+        options = main_line[1:]
+        title   = ""
+        if len(lines) > 1:
+            title = lines[1]
+        vote_options = []
+        if len(lines) > 2:
+            vote_options = lines[2:]
+
+        if 0:
+            print("All : %s" % (lines))
```

Comment:
You might want to use the logger to log these instead of print so that the output could be disabled by passing `--quiet` to `run.py`